Ryanjung application pages

rymaju opened the pull request.

jackblanc commented on src/components/Apply/Apply.vue:
```diff
@@ -0,0 +1,162 @@
+      this.resume = this.$refs.file.files[0];
+    },
+    isEmptyString(string) {
+      return string.trim().length === 0;
```
you should be able to do `string === ""`

wwohlers replied:
I think you might have to do `string.trim() === ""` to ensure that you don't count whitespace

jackblanc commented on src/components/Apply/Apply.vue:
```diff
@@ -0,0 +1,162 @@
+
+      let isValidForm = true;
+
+      if (this.isEmptyString(this.name)) {
```
I think it'd be cleaner if instead of having all these methods you just did the checks inline. But if more complex validation was needed later on this would be the right pattern

jackblanc commented on src/components/Contact/Contact.vue:
```diff
@@ -0,0 +1,126 @@
+      }
+      return isValidForm;
+    },
+    apply(formdata) {
```
what's the thinking behind this?

jackblanc replied:
ok, so I see the idea is that this is a submit function with nowhere to send data?

jackblanc commented on src/components/Contact/Contact.vue:
```diff
@@ -0,0 +1,126 @@
+    submit() {
+      this.submitted = false;
+      if (this.validateForm()) {
+        this.apply({
```
generally we should only use console.log for debugging. But considering we don't have anywhere to send the data yet this is fine for now.

rymaju commented:
- Added jest (by completely overwriting .babelrc) as well as adding tests for almost all methods in both Apply and Contact components. One method, handleFileUpload(), was not tested as mocking the file upload was a lot of work for not a lot of gain (the code in the function is a one liner anyway, not exactly a complex logic). Note that there is a lot of similarity in the methods of Apply and Contact that may be best separated into a "forms.js" file later, or possibly a parent component.

- Added css styling to center both forms and added some styling to input, select, textarea, and buttons as well. Added media query for a max width of 600px, so that the view does not break on mobile.

- Finally, jest coverage was removed and git ignored.
